scipy/sparse/base.py

```diff
@@ -386,6 +386,16 @@ def __rmul__(self, other):  # other * self
                 tr = np.asarray(other).transpose()
             return (self.transpose() * tr).transpose()
 
+    #####################################
+    # matmul (@) operator (Python 3.5+) #
+    #####################################
+
+    def __matmul__(self, other):
+        return self.__matmul__(other)
```

Comment:
I suppose you meant `__mul__` for the implementation. this is an infinite regress.